index.js

```diff
@@ -28,13 +28,26 @@ function createError (code, message, statusCode = 500, Base = Error) {
     }
 
     this.statusCode = statusCode || undefined
+    this.uriReference = uriReference || undefined
   }
   FastifyError.prototype[Symbol.toStringTag] = 'Error'
 
   FastifyError.prototype.toString = function () {
     return `${this.name} [${this.code}]: ${this.message}`
   }
 
+  FastifyError.prototype.toRFC7807 = function (instance, details) {
+    return {
+      type: this.uriReference || 'about:blank',
```

Comment:
I think the URI reference should be taken from the request itself and defined at the top for all errors of the same kind.

Reply:
I called it uriReference because according to RFC7807 it is called type. But that would be imho confusing.

https://datatracker.ietf.org/doc/html/rfc7807#section-3.1

```
A URI reference [RFC3986] that identifies the
     * problem type.  This specification encourages that, when
     * dereferenced, it provide human-readable documentation for the
     * problem type (e.g., using HTML [W3C.REC-html5-20141028]).
     * When this member is not present, its value is assumed to be
     * "about:blank".
```

So e.g. we create a FastifyError with code 'FST_ERR_BAD_URL'  then the uriReference would be `https://www.fastify.io/docs/latest/Reference/Errors/#fst_err_bad_url`

Reply:
Thats why this is an parameter of the createError function and not on the toRFC7807. On the toRFC7807 I use the instance parameter to provide uri from the request. 

I wanted to ensure that we have no potential backwards breaking, but I could put instance and details as parameters of the actual Error. So toRFC7807 would just this.instance and this.details instead using them from toRFC7807.
Maybe that is better, as we would have those values at the time we throw the Error?